File: a006. 一元二次方程式/C/Lyu.c
#include <stdio.h>
#include <math.h>
int main(void)
{
    int a, b, c, proof, x1, x2;
    scanf("%d %d %d", &a, &b, &c);
    proof = b * b - 4 * a * c;
    if (proof < 0)
        printf("No real root");
    else
    {
        x1 = (-b + sqrt(proof)) / (2 * a);
        x2 = (-b - sqrt(proof)) / (2 * a);
        if (x1 == x2)
            printf("Two same roots x=%d", x1);
        else
            printf("Two different roots x1=%d , x2=%d", x1, x2);
    }
}

File: a006. 一元二次方程式/Java/Lyu.java
import java.util.Scanner;

public class Lyu {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int a = scan.nextInt(), b = scan.nextInt(), c = scan.nextInt();
        int judgmental = b * b - 4 * a * c;
        if (judgmental < 0) {
            System.out.println("No real root");
        } else {
            double x1 = (-b + Math.sqrt(judgmental)) / (2 * a);
            double x2 = (-b - Math.sqrt(judgmental)) / (2 * a);
            System.out.println((judgmental == 0) ? "Two same roots x=" + Integer.toString((int) x1)
                    : "Two different roots x1=" + Integer.toString((int) x1) + " , x2=" + Integer.toString((int) x2));
        }
        scan.close();
    }
}